### PR TITLE
make docker-entrypoint exit on error

### LINKED ISSUE #47414: Check for db:prepare success before starting rails server via Dockerfile
### Steps to reproduce

Deploy a Rails 7.1 application using the provided dockerfile and a misconfigured database

### Expected behavior

Deployment should halt

### Actual behavior

Server is launched without a database

### System configuration
**Rails version**: 7.1

**Ruby version**: 3.2.1

### Recommended solution

Add `-e` to the shebang line, like so: https://github.com/rubys/dockerfile-rails/commit/26c66247cf93adfb42240bbd8a1ed49c9fe6e3b6

### PR DESCRIPTION
**Pull request by rubys**
Fixes #47414

cc: @zzak @byroot @dhh

### CODE REVIEW

**Comment by dhh** [comment]
What's the error the user is shown? Or you'll just have to check docker logs?

**Comment by rubys** [comment]
A lot will depend on your deployment platform.  What happened on fly.io is that a new VM was spun up, the migration failed, but no indication was given so traffic was routed to the new VM and and the old VM was shut down.

**Comment by dhh** [comment]
anyway, sounds good to me that we fail to boot if we fail to complete. fail early.

**Comment by rubys** [comment]
I should add: with fly.io if you fail early, a tail of the logs will be shown to the person running `fly deploy`.  What happened in this case is that the docker entrypoint appeared to succeed, health checks passed, so no logs were shown and it appeared to the developer that the migration was not run at all.  Failing at the point of the error results in a much better developer experience.

File: railties/lib/rails/generators/rails/app/templates/docker-entrypoint.tt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -e
```
Comment (zzak) [inline railties/lib/rails/generators/rails/app/templates/docker-entrypoint.tt:1]: Does this pass shellcheck? I thought they prefer using `set` in separate statements, but I could be wrong. Will investigate. 👀

**Pull request merged**